src/vs/workbench/contrib/chat/browser/chatInputPart.ts

```diff
@@ -563,11 +565,29 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
 		this.navigateHistory(false);
 	}
 
-	private navigateHistory(previous: boolean): void {
+	private async navigateHistory(previous: boolean): Promise<void> {
 		const historyEntry = previous ?
 			this.history.previous() : this.history.next();
 
-		const historyAttachments = historyEntry.state?.chatContextAttachments ?? [];
+		let historyAttachments = historyEntry.state?.chatContextAttachments ?? [];
+
+		// Check for images in history to restore the value.
+		if (historyAttachments.length > 0) {
+			historyAttachments = await Promise.all(historyAttachments.map(async (attachment) => {
+				if (attachment.isImage && attachment.references?.length && URI.isUri(attachment.references[0].reference)) {
+					try {
+						const buffer = await this.fileService.readFile(attachment.references[0].reference);
+						const newAttachment = { ...attachment };
+						newAttachment.value = buffer.value.buffer;
```

Comment:
Can you send the URI now instead of reading it here and transferring the data?

Reply:
I think we can only do this for the pasted image scenario (the pasted image's URI points to a scaled image based on the uint8array from teh data transfer, but a drag and dropped or attached image's URI will be the original file, not the scaled image)